### ISSUE TITLE
ui test: add incremental path forward to supporting more tests on new platforms

### ISSUE THREAD

**Issue by oli-obk** [rust-lang/miri#2324]
We could add a filter-list file that ui_test will use to choose whether to run a test, and also add a flag that makes sure we run all tests and error if any tests not on the list pass. We could even bless the list file automatically. Mostly I want it so CI detects any changes for partially supported targets.

though it would mean we always run *all* tests for a target, even those it can't support.

Similarly for `// ignore-X` tests, we could run them there and mention if they pass under a special mode.

Very rambly post, but I think there's something here we could explore

**Comment by RalfJung** [comment]
I am honestly not entirely sure if we want this... when a new test is being added, these 'minimal' targets should not be in the way.

> Mostly I want it so CI detects any changes for partially supported targets.

Why that? What's important is that the few things that work keep working, but when a new test is being added, I don't think we want to force the author of that PR to acknowledge whether or not it works on the partially supported target. That's just extra work for them, whereas the point of these partially supported targets is that we don't fully support them yet and hence they shouldn't cause friction when developing other parts of Miri.

**Comment by oli-obk** [comment]
well, at the very least we could support a file per unsupported target that lists all the passing tests that should keep passing (and make it easy to bless that file). This way the work is completely on the target maintainer.

**Comment by RalfJung** [comment]
Yeah, I guess that would be a slightly more elaborate form of the `run_tests_minimal` list of tests we currently have.
